index.js

```diff
@@ -47,6 +49,26 @@ class HyperDHT extends DHT {
       this._peers.destroy()
       this._store.clear()
     })
+
+    if (opts.adaptive) {
+      if (this.ephemeral !== true) {
+        this.destroy()
+        throw Error('adaptive mode can only applied when ephemeral: true')
+      }
+      this.once('ready', () => {
+        const timeout = setTimeout(() => {
+          this.joinDht((err) => {
+            if (err) {
+              err.message = `Unable to dynamically become non-ephemeral: ${err.message}`
+              this.emit('warning', err)
+              return
+            }
+            this.emit('dynamically-non-ephemeral')
+          })
+        }, ADAPT_EPHEMERALITY_AFTER)
+        timeout.unref()
```

Comment:
we should clear the timeout on destroy

Reply:
oh and we should check if we're holepunchable also before going persistent

Reply:
done!